### ISSUE TITLE
Custom layer created with no parameter

### ISSUE THREAD

**Issue by quocdat32461997** [keras-team/keras#13939]
When I create the Multi-Head-Attention layer below similar to the following link [Tensorflow Transformer](https://www.tensorflow.org/tutorials/text/transformer#masking) 
`class MultiHeadAttention(keras.layers.Layer):
    def __init__(self, d_model, num_heads, trainable = False, name = 'MHA', **kwargs):
        self.num_heads = num_heads
        self.d_model = d_model
        self.trainable = trainable
        self.name = name
    
        assert d_model % self.num_heads == 0
        self.depth = self.d_model // self.num_heads
        super(MultiHeadAttention, self).__init__(**kwargs)
        
    def build(self, input_shape):
        
        self.wq = Dense(self.d_model, activation = 'relu', trainable = self.trainable)
        self.wk = Dense(self.d_model, activation = 'relu', trainable = self.trainable)
        self.wv = Dense(self.d_model, activation = 'relu', trainable = self.trainable)
    
        self.dense = keras.layers.Dense(self.d_model, trainable = self.trainable)
        
        super(MultiHeadAttention, self).build(input_shape)
    
    def _scaled_dot_product_attention(self, q, k, v, mask = None):
        """Calculate the attention weights.
        q, k, v must have matching leading dimensions.
        k, v must have matching penultimate dimension, i.e.: seq_len_k = seq_len_v.
        The mask has different shapes depending on its type(padding or look ahead) 
        but it must be broadcastable for addition.

            Args:
                q: query shape == (..., seq_len_q, depth)
                k: key shape == (..., seq_len_k, depth)
                v: value shape == (..., seq_len_v, depth_v)
                mask: Float tensor with shape broadcastable to (..., seq_len_q, seq_len_k). Defaults to None.

            Returns:
                output, attention_weights
            """
        matmul_qk = tf.matmul(q, k, transpose_b=True)  # (..., seq_len_q, seq_len_k)
      
        #matmul_qk = K.dot(q, k)
        # scale matmul_qk
        dk = tf.cast(tf.shape(k)[-1], tf.float32)
        scaled_attention_logits = matmul_qk / tf.math.sqrt(dk)
        
        # add the mask to the scaled tensor.
        if mask is not None:
            scaled_attention_logits += (mask * -1e9)  

        # softmax is normalized on the last axis (seq_len_k) so that the scores
        # add up to 1.
        attention_weights = tf.nn.softmax(scaled_attention_logits, axis=-1)  # (..., seq_len_q, seq_len_k)
        output = tf.matmul(attention_weights, v)  # (..., seq_len_q, depth_v)
        return output, attention_weights
        
    def split_heads(self, x, batch_size):
        """Split the last dimension into (num_heads, depth).
        Transpose the result such that the shape is (batch_size, num_heads, seq_len, depth)
        """
        #x = tf.reshape(x, (batch_size, -1, self.num_heads, self.depth))
        x = tf.reshape(x, (batch_size, -1, self.num_heads, self.depth))
        return tf.transpose(x, perm=[0, 2, 1, 3])
    """
    def build(self, input_shape):
        #Create trainable weight varaible for the Multi-Head-Attention Layer
        self.kernel = self.add_weight(name = 'kernel', shape = (input_shape[0][1], self.output_dim), \
                                     initialier = 'uniform', trainable = self.trainable)
        super(MultiHeadAttention, self).build(input_shape)
    """
    
    def call(self, inputs, mask = None):
        batch_size = tf.shape(inputs)[0]
    
        q = self.wq(inputs)  # (batch_size, seq_len, d_model)
        k = self.wk(inputs)  # (batch_size, seq_len, d_model)
        v = self.wv(inputs)  # (batch_size, seq_len, d_model)
    
        q = self.split_heads(q, batch_size)  # (batch_size, num_heads, seq_len_q, depth)
        k = self.split_heads(k, batch_size)  # (batch_size, num_heads, seq_len_k, depth)
        v = self.split_heads(v, batch_size)  # (batch_size, num_heads, seq_len_v, depth)
    
        # scaled_attention.shape == (batch_size, num_heads, seq_len_q, depth)
        # attention_weights.shape == (batch_size, num_heads, seq_len_q, seq_len_k)
        scaled_attention, attention_weights = self._scaled_dot_product_attention( q, k, v, mask)
    
        scaled_attention = tf.transpose(scaled_attention, perm=[0, 2, 1, 3])  # (batch_size, seq_len_q, num_heads, depth)

        concat_attention = tf.reshape(scaled_attention, (batch_size, -1, self.d_model))  # (batch_size, seq_len_q, d_model)

        output = self.dense(concat_attention)  # (batch_size, seq_len_q, d_model)
   
        return output
`
I got the no parameter for the Multi-Head-Attention-Layer when adding the custom layer by either Sequential or stacking with Input layer from pure Keras. I am using Tensorflow 2.1.0 and Keras 2.3.1 on AMD Radeon VII
![Screenshot from 2020-04-06 02-35-44](https://user-images.githubusercontent.com/18373768/78533945-5b1f9500-77af-11ea-9878-610ed41385b2.png)

**Comment by sushreebarsa** [comment]
@quocdat32461997 It looks like you are using an older Version of Tensorflow . Could you please execute your code using Latest Version 2.6.0 and let us know if the issue still persists? Thanks!
